### PR TITLE
add support for freebsd

### PR DESCRIPTION
**Pull request by sethlyons**

### CODE REVIEW

**Comment by sethlyons** [comment]
i added two additional commits here that were unrelated to my change but required to make the tests pass.  if you want i can pull them out of here and submit them in a separate PR.

**Comment by traylenator** [comment]
@sethlyons seems good  and thanks.
Could you update the metadata file for the new OS.

**Comment by sethlyons** [comment]
all set.  i alphabetized the OSes as well.

File: .rubocop.yml [hunk 1]
```diff
@@ -302,7 +302,7 @@ Style/EmptyLiteral:
 Metrics/LineLength:
   Enabled: False
 
-Style/MethodCallParentheses:
+Style/MethodCallWithoutArgsParentheses:
```
Comment (alexjfisher) [inline .rubocop.yml]: This file should only be modified via a modulesync.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -24,8 +24,8 @@
     puppetversion: Puppet.version,
     facterversion: Facter.version
   }
-  default_facts.merge!(YAML.load(File.read(File.expand_path('../default_facts.yml', __FILE__)))) if File.exist?(File.expand_path('../default_facts.yml', __FILE__))
-  default_facts.merge!(YAML.load(File.read(File.expand_path('../default_module_facts.yml', __FILE__)))) if File.exist?(File.expand_path('../default_module_facts.yml', __FILE__))
+  default_facts.merge!(YAML.safe_load(File.read(File.expand_path('../default_facts.yml', __FILE__)))) if File.exist?(File.expand_path('../default_facts.yml', __FILE__))
```
Comment (alexjfisher) [inline spec/spec_helper.rb]: This file should only be modified via a modulesync.
Looks like this is fixed in the latest release.  See https://github.com/voxpupuli/modulesync_config/pull/308

**Comment by alexjfisher** [comment]
https://github.com/voxpupuli/puppet-squid/pull/46 needs fixing up and merging then this PR can be rebased and the 2 rubocop commits won't be needed.

**Comment by bastelfreak** [comment]
Hi @sethlyons, thanks for the PR. Can you rebase + drop the two rubocop commits please? Then I can merge your PR and do a new release.

**Comment by sethlyons** [comment]
all set!

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -41,8 +41,15 @@
               it { is_expected.to contain_concat_fragment('squid_header').with_content(%r{^access_log\s+daemon:/var/log/squid/access.log\s+squid$}) }
             end
           end
+        when 'FreeBSD'
```
Comment (alexjfisher) [inline spec/classes/init_spec.rb]: Should have probably created a new context here too.

Comment (sethlyons) [inline spec/classes/init_spec.rb]: agreed.  apologies...i should have realized that.  do you want a new PR/branch for that?

Comment (alexjfisher) [inline spec/classes/init_spec.rb]: @sethlyons If you've got time, that'd be great.  Thanks.

**Pull request merged**